hal/armv7m/cpu.c

```diff
@@ -19,14 +19,20 @@
 #include "spinlock.h"
 #include "string.h"
 
-#ifdef CPU_STM32
+
+#if defined(CPU_STM32L152XD) || defined(CPU_STM32L152XE) || defined(CPU_STM32L4X6)
 #include "stm32.h"
 #endif
 
-#ifdef CPU_IMXRT
-#include "imxrt.h"
+#if defined(CPU_IMXRT105X) || defined(CPU_IMXRT106X)
+#include "imxrt10xx.h"
 #endif
 
+#ifdef CPU_IMXRT117X
+#include "imxrt117x.h"
+#endif
+
+
```

Comment:
This logic could be hidden inside some header file. mcu.h maybe? This would make this #ifs to be in one place only

Reply:
This logic should be implemented only in hal.h. I agree with you that #ifs should be located in one place only.